### ISSUE TITLE
Consider restricting requirement mapping

### ISSUE THREAD

**Issue by WilcoFiers** [w3c/wcag-act#290]
Proposal was to restrict the accessibility requirement mapping to a finite list, such as:

- Passed: The requirement(s) are satisfied
- Passed: Further testing is necessary to determine if the success criteria are satisfied
- Inapplicable: The requirement(s) are satisfied
- Inapplicable: Further testing is necessary to determine if the success criteria are satisfied
- Failed: The requirement(s) are *not* satisfied
- Failed: Further testing is necessary to determine if the success criteria are satisfied

**Comment by WilcoFiers** [comment]
I have two objections to restricting the requirement mapping in this way.

1. It forces rule authors to put aggregation logic into expectations. My concern with this is that expectations are written to get passed / failed outcomes for individual test targets, and not for aggregating the outcome for the page / test subject as a whole.

2. It opens back up the discussion of having composite rules use results of other composite rules. This idea was initially rejected because it would allow writing rules as a decision tree. While decision trees are certainly used in several accessibility tools, there are many tools that have a fairly flat structure and could not implement rules expressed in a decision tree. For that same reason the original procedural approach was also rejected, because tools using decision trees can't implement the procedural approach.

----

### Proposal

I do think flexibility in the accessibility requirements mapping is necessary to avoid the problems outlined above. But there is space between: "pick one of these" six, and "free form write how to aggregate". I would suggest that we add the following:

1. The accessibility requirements mapping MUST be objective. This ensures it can be automated.
2. Only the following information can be used to determine if the accessibility requirement was satisfied or not, or if further testing is needed:
	- The outcome of the test subject
	- The outcome of each test target
	- Whether or not content is embedded on the page using technologies other than those listed in test aspects.

**Comment by annethyme** [comment]
Other suggestion: 

Accessibility requirements: 
- 3.3.2 # Labels and Instructions (Full test)
- 4.1.2 # Name, Role, Value (Partial test)

**Comment by annethyme** [comment]
Possible mapping patterns, as Anne sees it: 
* “Input data” 
   * Rule in itself does not even map to a best practice (e.g. a rule that tests that `<img>` has `alt=""`, but this is only used as input in the rule that tests if images have an accessible name)
* Best practice test
   * Output cannot be mapped to satisfied / not satisfied for accessibility requirement
* Non-conformance test
   * Failed = not satisfied 
* Full test for the requirement 
   * Failed = not satisfied 
   * Passed = satisfied

Suggested placeholder to use when "Accessibility Requirements" section is empty: 
“For [my standard] this rule in itself is not an accessibility requirement”

**Comment by annethyme** [comment]
Use cases: 

- Best practice: Headings are properly nested
- Non-conformance test: 
- Full test for the requirement:

**Comment by annethyme** [comment]
Shadi's mapping from the minutes: 

- Accessibility requirement: <reference to an accessibility standard> | Related provision: <reference to any testable statement>
- Pass condition: <satisfied | further testing needed>
- Fail condition: <not-satisfied | further testing needed>
- Inapplicable condition: <satisfied | not-satisfied | further testing needed>
- Partial test: <yes | no>

**Comment by annethyme** [comment]
Romain's unanswered questions for this: 
- For atomic rules, that are only partial tests for several different accessibility requirements, should there be a way to indicate this? 
- For nested rules, which accessibility mapping takes the precedence? The top-most one in the hierarchy? Needs to be specced.

**Comment by annethyme** [comment]
Another use case: 

Composite rule: https://auto-wcag.github.io/auto-wcag/rules/SC2-1-2-no-keyboard-trap.html
Atomic rule 1: https://auto-wcag.github.io/auto-wcag/rules/SC2-1-2-no-keyboard-trap-standard-navigation
Atomic rule 2: https://auto-wcag.github.io/auto-wcag/rules/SC2-1-2-no-keyboard-trap-non-standard-navigation

For the atomic rules, the authors have expressed a wish to link to the composite rule, since this creates context for how to interpret the rules. 

For Atomic rule 1: https://auto-wcag.github.io/auto-wcag/rules/SC2-1-2-no-keyboard-trap-standard-navigation it would be nice to indicate this as a Best practice, since this is the preferred way of satisfying the success criterion.

For Atomic rule 2: https://auto-wcag.github.io/auto-wcag/rules/SC2-1-2-no-keyboard-trap-non-standard-navigation we would NOT want to indicate this as a Best practice.

**Comment by cpandhi** [comment]
How the rule impacts the accessibility requirement is useful information IMO.
+1 to Shadi's mapping with Ann's edits.

Should it also have, 
Pass condition: <satisfied | No further testing needed>
This could apply to failure techniques.

One way to map can be to sufficient techniques of a SC. 
For example: a rule that tests that <img> has alt="short description" maps to [H37](https://www.w3.org/TR/WCAG20-TECHS/H37.html) which satisfies Success Criterion 1.1.1 (Non-text Content)
